### ISSUE TITLE
Backup wallet config before overwriting beacon keys

### ISSUE THREAD

**Issue by denravonska** [gridcoin-community/Gridcoin-Research#330]
```
<Quezacoatl> ravon is there a possibility to automatically backup the .conf when sending a beacon? many peoples .confs get corrupted and they do not have any backup
<@Quezacoatl> You wont need to set up everything from scratch then
```

It should be a one-liner before storing the keys.

**Comment by Quezacoatl1** [comment]
And if they send another beacon that does not match the same keys, could we backup to another time just to be sure both keypairs are safe? :)

**Comment by denravonska** [comment]
@Quezacoatl1 This would work, right?

```
def AdvertiseBeacon:
   GenerateKeys()
   if SendBeacon() == OK:
      BackupConfig()
      SaveKeys()
```

**Comment by Quezacoatl1** [comment]
@denravonska haha I have no clue as a non-coder here, sorry. I can test it if you send me a .exe

Does BackupConfig() overwrite existing files with the same name?

**Comment by denravonska** [comment]
hmm, need to check how it names the backups but it does keep multiple copies of them.

**Comment by iFoggz** [comment]
Great idea. When BackupGridcoinWallet is called it has the date added to the filenames to be used but in the end for config it calls BackupConfigFile to write the backup you just have to push forward a path and filename if u want to customize it :)

But we should have a way to make the file unique at the time. BackupGridcoinWallet currently just does date and if it happens to be same date 900 blocks later it will overwrite the existing one.

Perhaps we can do like gridcoinresearch-beaconpublickey.conf then we wouldn't have to worry about an overwrite (which could overwrite the valid good keys). then if we say your public key says in beaconstatus is this they can find that file and make a copy of it into there main folder as gridcoinresearch.conf.
example: gridcoinresearch-HEXPUBLICKEY.conf. But this could cause another issue for when the beacon is re-advertised after 5 months using the same keys? maybe add if beacon older then 5 months don't do backup of config during beacon advertise if the file exsists as stated above or allow the overwrite. 

Just my ideas. We could use time but trying to make this easiest for getting them the right config.

**Comment by iFoggz** [comment]
I'm working on this at the moment.

**Comment by iFoggz** [comment]
Also note that fix #334 prevents the existing keys from being overwritten when valid to begin with so this is ready for closing.

**Comment by denravonska** [comment]
Closed by #361.
